### ISSUE TITLE
Request data preprocessing code

### ISSUE THREAD

**Issue by Novo-caine** [mingyangliu1024/ACON#4]
Could you please provide the HHAR_D and EMG data preprocessing code? Thanks

**Comment by mingyangliu1024** [comment]
Apologies for the delay, and thanks for your attention to our work. 

We follow the domain split of the original datasets (EMG has 4 domains and HHAR_D has 5 domains). For each domain, we randomly split the train set (80%) and the test set (20%) and save them as *.pt files. For convenience, the partitioned datasets can be downloaded from [EMG](https://drive.google.com/file/d/1RgO8gtJSecdqGyir7RamjjqbcesFsFce/view?usp=sharing) and [HHAR_D](https://drive.google.com/file/d/17CBFtmJpxXNyxgA5HH7gB0JURdJdIA04/view?usp=sharing).

We apologize that the download links of EMG and HHAR_D previously provided did not include the train/test splits. The links have now been updated.

Here we provide the random splitting code for HHAR_D for reference. The same applies to EMG, but the original file format is .npy.





```
      import h5py
      import torch
      import os
      import numpy as np
      
      def process_h5():
          h5_path = 'original_HHAR_woods/HAR.h5'
          hdf = h5py.File(h5_path, 'r')
          pt_path = 'HHAR_woods/'
          for index, key in enumerate(hdf.keys()):
              print(hdf[key].name, end=' ')
              print(type(hdf[key]['data']), end=' ')
              print(hdf[key]['data'].shape, end=' ')
              print(hdf[key]['labels'].shape)
      
              data = torch.from_numpy(hdf[key]['data'][:]).to(torch.float64)
              targets = torch.from_numpy(hdf[key]['labels'][:]).to(torch.float64)
      
              length = data.size(0)
              random_indices = torch.randperm(length)
              shuffled_data = data[random_indices]
              shuffled_data = torch.transpose(shuffled_data, 1, 2)
              shuffled_targets = targets[random_indices].squeeze()
              print(shuffled_data.shape)
              print(shuffled_targets.shape)
              
              train_split = round(length * 0.8)
      
              torch.save(
                  {'samples': shuffled_data[0:train_split],
                  'labels': shuffled_targets[0:train_split],},
                  os.path.join(pt_path, 'trian_{}.pt'.format(str(index)))
              )
              print('Train {}'.format(train_split))
              torch.save(
                  {'samples': shuffled_data[train_split:],
                  'labels': shuffled_targets[train_split:],},
                  os.path.join(pt_path, 'test_{}.pt'.format(str(index)))
              )
              print('Test {}'.format(length-train_split))
      
```
